app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt

```diff
@@ -115,6 +126,45 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
             }
       }
 
+  private fun searchLocation(query: String) =
+      viewModelScope.launch(Dispatchers.IO) {
+        val client = OkHttpClient()
+        val url = "https://nominatim.openstreetmap.org/search?q=$query&format=json&limit=4"
+        val request = Request.Builder().url(url).build()
+        _searchResponse.value =
+            try {
+              val response = client.newCall(request).execute()
+              val responseBody = response.body()?.string()
+              responseBody
+            } catch (e: Exception) {
+              e.printStackTrace()
+              "error"
+            }
+      }
+
+  fun getSearchLocation(query: String, time: Int) =
+      viewModelScope.launch {
+        searchLocation(query).join()
+        val gson = Gson()
+        if (_searchResponse.value == null) {
+          _searchResults.value = null
+        } else if (_searchResponse.value != null && _searchResponse.value != "") {
+          Log.d("searchLocation", _searchResponse.value!!)
+          val jsonArray = gson.fromJson(_searchResponse.value, JsonArray::class.java)
+          Log.d("searchLocation", jsonArray.toString())
+          _searchResults.value =
+              jsonArray.map { jsonString ->
+                val jsonObject = gson.fromJson(jsonString, JsonObject::class.java)
+                val displayName = jsonObject.get("display_name")?.asString ?: ""
+                val latitude = jsonObject.get("lat")?.asDouble ?: 0.0
+                val longitude = jsonObject.get("lon")?.asDouble ?: 0.0
```

Comment:
I'm not familiar with the `gson` library, can this throw an exception ? If the format is not the one expected we might want to display an error message instead of setting default values

Reply:
As openStreetMap always send a JsonArray as a string in case of a right response, there should be no error there. All the possible errors would happen during searchLocation